lib/RenderApp/Controller/RenderProblem.pm

```diff
@@ -523,22 +523,19 @@ sub generateJWTs {
 
 sub insert_mathquill_responses {
     my ( $form_data, $pg ) = @_;
-    for my $answerLabel ( keys %{ $pg->{pgcore}->{PG_ANSWERS_HASH} } ) {
-        my $mq_opts = $pg->{pgcore}->{PG_ANSWERS_HASH}->{$answerLabel}
-          ->{ans_eval}{rh_ans}{mathQuillOpts};
-        my $response_obj =
-          $pg->{pgcore}->{PG_ANSWERS_HASH}->{$answerLabel}->response_obj;
+    for my $answerLabel ( keys %{ $pg->{pgcore}{PG_ANSWERS_HASH} } ) {
+        my $mq_opts = $pg->{pgcore}{PG_ANSWERS_HASH}{$answerLabel}{ans_eval}{rh_ans}{mathQuillOpts} // '';
+        next if ( $mq_opts =~ /\s*disabled\s*/ );
```

Comment:
these `/s*` aren't doing anything for this regex... do we expect a word boundary for disabled? Perhaps `\s+`?

I did a cursory search and didn't find any documentation on mq_opts - do we have the details anywhere?

Reply:
You are correct. Either those shouldn't be there or the regular expression should be anchored. I think that I intended is for the regular expression to be anchored. I.e., with ^ at the beginning and $ at the end.

Reply:
`=~ ^disabled$` is equivalent to `== 'disabled'`, no? Maybe we want to include the `i` flag?

Reply:
Although this is where the \s*'s come in. But we could be more strict and not allow spaces.

Reply:
Also I think you meant `eq`

Reply:
Oh yeah, getting that perl-brain/js-brain scrambled again.

I guess my main issue here is that I don't fully grasp the required structure for `mathQuillOpts` -- is it just `key1: value1; key2: value2;`? And `disabled` is completely separate from that structure?

Reply:
Yeah, basically it is either full JSON encoded into a string or the string 'disabled'.

Reply:
Okay, I'm gonna merge all this then -- anchoring the regex can happen whenever...